bugbug/model.py

```diff
@@ -300,24 +379,33 @@ def classify(
             explainer = shap.TreeExplainer(self.clf)
             shap_values = explainer.shap_values(X)
 
-            # TODO: Actually implement feature importance visualization for multiclass problems.
-            if isinstance(shap_values, list):
-                shap_values = np.sum(np.abs(shap_values), axis=0)
+            if not isinstance(shap_values, list):
+                shap_values = [shap_values]
 
-            top_importances = self.get_important_features(
+            important_features = self.get_important_features(
                 importance_cutoff, shap_values
             )
 
+            # Workaround: handle multi class case for force_plot to work correctly
+            if not len(classes[0]) == 2:
+                pred_class_index = classes.argmax(axis=-1)[0]
+                explainer.expected_value = explainer.expected_value[0]
+            else:
+                pred_class_index = 0
+
             top_indexes = [
-                int(index) for importance, index, is_positive in top_importances
+                int(index)
+                for importance, index, is_positive in important_features["classes"][
+                    f"class {pred_class_index}"
+                ][0]
             ]
 
             feature_names = self.get_feature_names()
 
             with io.StringIO() as out:
                 p = shap.force_plot(
                     explainer.expected_value,
-                    shap_values[:, top_indexes],
+                    shap_values[0][:, top_indexes],
```

Comment:
Shouldn't this be shap_values[pred_class_index]?

Reply:
Traceback (most recent call last):
  File "run.py", line 165, in <module>
    main(parse_args(sys.argv[1:]))
  File "run.py", line 102, in main
    bug, probabilities=True, importances=True
  File "/home/harshit/Documents/bugbug/bugbug/model.py", line 398, in classify
    shap_values[pred_class_index][:, top_indexes],
IndexError: too many indices for array